applications/FluidDynamicsApplication/custom_strategies/schemes/velocity_bossak_adjoint_scheme.h

```diff
@@ -0,0 +1,417 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_VELOCITY_BOSSAK_ADJOINT_SCHEME_H_INCLUDED)
+#define KRATOS_VELOCITY_BOSSAK_ADJOINT_SCHEME_H_INCLUDED
+
+// System includes
+#include <vector>
+#include <string>
+#include <unordered_set>
+#include <functional>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/checks.h"
+#include "includes/kratos_parameters.h"
+#include "solving_strategies/schemes/scheme.h"
+#include "response_functions/adjoint_response_function.h"
+#include "utilities/variable_utils.h"
+#include "utilities/indirect_scalar.h"
+#include "utilities/adjoint_extensions.h"
+#include "utilities/parallel_utilities.h"
+#include "solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h"
+
+// Application includes
+#include "custom_utilities/fluid_adjoint_utilities.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+template <unsigned int TDim, class TSparseSpace, class TDenseSpace, unsigned int TBlockSize = TDim + 1>
```

Comment:
Why templating with dimension (block size)? I'd be consistent with the core and application schemes.

Reply:
done :)